src/services/dataSource.js

```diff
@@ -0,0 +1,43 @@
+export const getDataSources = (appId) => {
+  const getOptions = appId ? {appId: appId} : {};
+
+  return Fliplet.DataSources.get(getOptions);
+};
+
+export const getDataSource = (dataSourceId) => {
+  return Fliplet.DataSources.getById(dataSourceId);
+};
+
+export const createDataSource = (widgetData) => {
+  return Fliplet.Modal.prompt({
+    title: 'Enter a name for the data source',
+    value: widgetData.default.name
+  })
+    .then(dataSourceName => {
+      if (dataSourceName === null) {
+        return false;
+      }
+
+      if (!dataSourceName) {
+        Fliplet.Modal.alert({
+          message: 'Data source name can\'t be empty. Plaese enter data source name again.'
+        })
+          .then(() => {
+            createDataSource();
+          });
```

Comment:
This doesn't seem to stop the code from creating a data source with empty string as name.

Reply:
@tonytlwu in the `createDataSource` function we do check if the data source name empty. This function is placed in the `src/services/dataSource.js` file.

Reply:
It does check if the data source name is empty. I can see that in the `if (!dataSourceName)` line. But your code doesn't stop the `Fliplet.DataSources.create()` line from executing if there is no data source name.

Reply:
@tonytlwu Yes, you are right. Will it be okay if we change it to `return Fliplet.Modal.alert` or better go with `if ... else`?

Reply:
`return Fliplet.Modal.alert()` works